contracts/auto-withdrawer/src/contract.rs

```diff
@@ -177,34 +166,29 @@ fn execute_unbond(
     info: MessageInfo,
     batch_id: Uint128,
 ) -> ContractResult<Response<NeutronMsg>> {
-    let bonding_id = get_bonding_id(info.sender.to_string(), batch_id);
+    let bonding_id = get_bonding_id(&info.sender, batch_id);
     let bonding = bondings_map().load(deps.storage, &bonding_id)?;
     ensure_eq!(info.sender, bonding.bonder, ContractError::Unauthorized {});
     bondings_map().remove(deps.storage, &bonding_id)?;
 
     let withdrawal_denom_prefix = WITHDRAWAL_DENOM_PREFIX.load(deps.storage)?;
     let withdrawal_token_address = WITHDRAWAL_TOKEN_ADDRESS.load(deps.storage)?;
-    let withdrawal_denom = get_full_withdrawal_denom(
-        withdrawal_denom_prefix,
-        withdrawal_token_address.to_string(),
-        batch_id,
-    );
-
-    let withdrawal_asset_msg: BankMsg = BankMsg::Send {
-        to_address: info.sender.clone().into_string(),
-        amount: vec![Coin::new(
-            bonding.withdrawal_amount.u128(),
-            withdrawal_denom,
-        )],
-    };
-
-    let deposit_msg: BankMsg = BankMsg::Send {
-        to_address: info.sender.clone().into_string(),
-        amount: bonding.deposit,
+    let withdrawal_denom =
+        get_full_withdrawal_denom(withdrawal_denom_prefix, withdrawal_token_address, batch_id);
+
+    let mut send_assets_vec = vec![Coin::new(
+        bonding.withdrawal_amount.u128(),
+        withdrawal_denom,
+    )];
+    send_assets_vec.extend(bonding.deposit);
```

Comment:
What are you going to do if `bonding.deposit` already contains `withdrawal_denom` coins? I am sure that current code will abort with Cosmos SDK complaining about malformed coins array.

Reply:
Fixed, made two messages

Reply:
Yep, I would rather choose between two inefficient bank messages or a single properly optimized one, rather then going for a buggy-optimized one :D

Should we mark this thread as resolved, or would you like to take time and write a properly optimized version?

Reply:
In the comment below, you said that processing coin vectors mappings is a complex feature but I can implement this

Reply:
Here it is a possible optimization which we could either take or leave behind, and it doesn't even involve working with vectors, I think, because what we have to merge is a single coin into a vector of coins. But then we will also have to sort a new vector. Bank messages are not super expensive so maybe it is not worth it.

Reply:
Seems that we have already the method we need. I fixed this